pkg/controller/podgc/gc_controller.go

```diff
@@ -124,6 +129,75 @@ func isPodTerminated(pod *v1.Pod) bool {
 	return false
 }
 
+// isPodTerminating returns true if the pod is terminating.
+func isPodTerminating(pod *v1.Pod) bool {
+	if phase := pod.Status.Phase; phase != v1.PodPending && phase != v1.PodRunning && phase != v1.PodUnknown {
+		return true
+	}
+	if pod.ObjectMeta.DeletionTimestamp != nil {
+		klog.Infof("garbage collecting pod %s that is terminating. Phase [%v]", pod.Name, pod.Status.Phase)
+		return true
+	}
+	return false
+}
+
+func (gcc *PodGCController) gcTerminating(pods []*v1.Pod) {
+	terminatingPods := []*v1.Pod{}
+	for _, pod := range pods {
+		if isPodTerminating(pod) {
+			node, err := gcc.nodeLister.Get(string(pod.Spec.NodeName))
+			if err != nil {
+				klog.Errorf("failed to get node:%s", err)
+				continue
+			}
+
+			// Add this pod to terminatedPods list only if the following conditions are met:
+			// 1. Node is not ready.
+			// 2. Node has `out-of-service` taint with `NoExecute` effect.
```

Comment:
In order to avoid misunderstanding, it would be better to write the well-known taint name.
```suggestion
			// 2. Node has `node.kubernetes.io/out-of-service` taint with `NoExecute` effect.
```